### PR TITLE
feature: initial bridge contract implementation

### PR DESCRIPTION
**Pull request by vigan-abd**

### CODE REVIEW

File: src/ThriveBridgeDestination.sol [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+            !mintNonces[sender][nonce],
+            "ThriveProtocol: request already processed"
+        );
+        mintNonces[sender][nonce] = true;
+
+        bytes32 hash = SignatureHelper.hashBridgeRequest(
+            address(this), sender, receiver, nonce, amount
+        );
+        bool validSig =
+            SignatureHelper.verifyBridgeRequest(sender, hash, signature);
```
Comment (lorentsinani) [inline src/ThriveBridgeDestination.sol:150]: can we use different name for this variable? In the first sight it did not look that understandable so maybe use `validSignature`

File: src/ThriveBridgeDestination.sol [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    ) internal virtual nonReentrant {
+        require(amount > 0, "ThriveProtocol: amount must be greater than zero");
+
+        uint256 nonce = burnNonces[sender];
+        burnNonces[sender]++;
+
+        bytes32 hash = SignatureHelper.hashBridgeRequest(
+            srcContract, sender, receiver, nonce, amount
+        );
+        bool validSig =
```
Comment (lorentsinani) [inline src/ThriveBridgeDestination.sol:184]: same here for validSig

File: src/ThriveIERC20Wrapper.sol [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        _;
+    }
+
+    function setMinter(address _minter) external onlyOwner {
+        minter = _minter;
+    }
+
+    function mint(address recipient, uint256 amount) external onlyMinter {
+        _mint(recipient, amount);
+    }
```
Comment (lorentsinani) [inline src/ThriveIERC20Wrapper.sol:34]: is there `_mint` function missing?

**Review by lorentsinani — APPROVED** [review]
LGTM just added that comment for all `validSig` variables across the smart contracts.

**Comment by vigan-abd** [comment]
test scenario

access control: https://abc-thrive.cloud.blockscout.com/address/0x8529531D460A2ad32FAE2002d569089440979cEE?tab=read_write_proxy
role: 0x0000000000000000000000000000000000000000000000000000000000000000

case 1: erc20-erc20
source erc20: https://abc-thrive.cloud.blockscout.com/address/0x771B8d05ed8005B5E5Ab6eE985A82017e68d99F8
dest erc20: https://abc-thrive.cloud.blockscout.com/address/0xD42E14B691Da958A29EA45420Ae794606A40b094
source contract: https://abc-thrive.cloud.blockscout.com/address/0x8B9d6118578CAe6B2970E7f0982cd8A564222cf4
dest contract: https://abc-thrive.cloud.blockscout.com/address/0xB4af7Ff2C9C94980402DE9Cb568974106f36435D
bridge 10 STK -> 10 DTK:
- https://abc-thrive.cloud.blockscout.com/tx/0x93129398b5425df1d469f75c66b8f6acd0e8805cae4465ff02ecd9c551863826?tab=logs
- https://abc-thrive.cloud.blockscout.com/tx/0x660c3fda32e8e51b919819e038c35b099622bdc41708251bac3c6840f655dee4?tab=logs
bridge 5 DTK -> 5 STK:
- https://abc-thrive.cloud.blockscout.com/tx/0xe4e96fcbdaa4194219c43c47991a70693043ee990b82eb008390192d37045e7f
- https://abc-thrive.cloud.blockscout.com/tx/0xbea378c41df8bab2e22799381b2e15f29c0fe2cb666490fdf0d6fdc6fbcc0268

**Pull request merged**